crates/mdns/src/dns/answer.rs

```diff
@@ -0,0 +1,185 @@
+#[allow(unused_imports)]
+use byteorder::{BigEndian, ByteOrder, ReadBytesExt, WriteBytesExt};
+use std::{default::Default, io::Cursor};
+
+/// Response answer
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Answer {
+    Unknown(Vec<u8>),
+    Data(AnswerSection),
+}
+
+/// Answer section of a DNS message packet.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct AnswerSection {
+    pub(crate) dn_len: u16,
+    pub(crate) domain_name: String,
+    // Should be CNAME = 5
+    pub(crate) answer_type: u16,
+    // = 1 ?
+    pub(crate) answer_class: u16,
+    pub(crate) ttl: u32,
+    pub(crate) data_len: u16,
+    pub(crate) data: Vec<Target>,
+}
+
+impl Default for AnswerSection {
+    fn default() -> Self {
+        Self {
+            dn_len: 0,
+            domain_name: String::default(),
+            answer_type: 5,
+            answer_class: 1,
+            ttl: 255,
+            data_len: 0,
+            data: Vec::new(),
+        }
+    }
+}
+
+impl AnswerSection {
+    /// New with default values for ttl(255), type(5) and class(1).
+    pub fn new(name: &str, targets: &[Target]) -> Self {
+        Self {
+            dn_len: name.len() as u16,
+            domain_name: name.to_owned(),
+            data_len: targets.len() as u16,
+            data: targets.to_vec(),
+            ..Default::default()
+        }
+    }
+
+    /// New with Time To Live value specified.
+    pub fn new_with_ttl(name: &str, targets: &[Target], ttl: u32) -> Self {
+        Self {
+            dn_len: name.len() as u16,
+            domain_name: name.to_owned(),
+            ttl,
+            data_len: targets.len() as u16,
+            data: targets.to_vec(),
+            ..Default::default()
+        }
+    }
+
+    /// Builds an [`AnswerSection`] from a byte cursor.
+    pub fn from_raw(mut cursor: &mut Cursor<&Vec<u8>>) -> MulticastDnsResult<Self> {
+        let dn_len = cursor.read_u16::<BigEndian>()?;
+        let mut domain_name: Vec<u8> = Vec::with_capacity(dn_len as usize);
+        for _ in 0..dn_len {
+            domain_name.push(cursor.read_u8()?);
+        }
+        let answer_type = cursor.read_u16::<BigEndian>()?;
+        let answer_class = cursor.read_u16::<BigEndian>()?;
+        let ttl = cursor.read_u32::<BigEndian>()?;
+        let data_len = cursor.read_u16::<BigEndian>()?;
+        let mut data = Vec::with_capacity(data_len as usize);
+
+        for _ in 0..data_len {
+            data.push(Target::from_raw(&mut cursor)?);
+        }
+
+        Ok(Self {
+            dn_len,
+            domain_name: String::from(std::str::from_utf8(&domain_name)?),
+            answer_type,
+            answer_class,
+            ttl,
+            data_len,
+            data,
+        })
+    }
+
+    pub fn write(&self, mut packet: &mut Vec<u8>) -> MulticastDnsResult<()> {
+        packet.write_u16::<BigEndian>(self.dn_len)?;
+
+        for byte in self.domain_name.as_bytes().to_vec() {
+            packet.write_u8(byte)?;
```

Comment:
We figured out this has to be not standard compliant because it does some weird stuff with our use cases like `wss://1.2.3.4:12345?a=HcMmymachineid` => `wss://1 | 2 | 3 | 4:12345?a=HcMmymachineid`